tests/task.test.ts

```diff
@@ -12,135 +12,223 @@ const index = {
+      expect(tasks.results[0].finishedAt).toBeDefined()
+      expect(tasks.results[0].startedAt).toBeDefined()
+    })
+
+    test(`${permission} key: Get all tasks with type filter`, async () => {
+      const client = await getClient(permission)
+      const task1 = await client.index(index.uid).addDocuments([{ id: 1 }])
+      await client.waitForTask(task1.taskUid)
+
+      const task2 = await client.index(index.uid).deleteDocument(1)
+      await client.waitForTask(task2.taskUid)
+
+      const tasks = await client.getTasks({
+        type: [
+          TaskTypes.DOCUMENTS_ADDITION_OR_UPDATE,
+          TaskTypes.DOCUMENT_DELETION,
+        ],
+      })
+      const onlyDocumentAddition = new Set(
+        tasks.results.map((task) => task.type)
       )
+
+      expect(onlyDocumentAddition.size).toEqual(2)
     })
 
-    test(`${permission} key: Try to get update that does not exist`, async () => {
+    test(`${permission} key: Get all tasks with status filter`, async () => {
       const client = await getClient(permission)
-      await expect(
-        client.index(index.uid).getTask(2545)
-      ).rejects.toHaveProperty('code', ErrorStatusCode.TASK_NOT_FOUND)
+      const enqueuedTask = await client
+        .index(index.uid)
+        .addDocuments([{ id: 1 }])
+      await client.waitForTask(enqueuedTask.taskUid)
+
+      const tasks = await client.getTasks({
+        status: [TaskStatus.TASK_SUCCEEDED, TaskStatus.TASK_FAILED],
+      })
+      const onlySuccesFullTasks = new Set(
+        tasks.results.map((task) => task.status)
+      )
+
+      expect(onlySuccesFullTasks.size).toEqual(2)
     })
-  }
-)
 
-describe.each([{ permission: 'Public' }])(
-  'Test on updates',
-  ({ permission }) => {
-    beforeEach(async () => {
-      await clearAllIndexes(config)
+    test(`${permission} key: Get all tasks with indexUid filter`, async () => {
+      const client = await getClient(permission)
+      const task1 = await client.index(index.uid).addDocuments([{ id: 1 }])
+      const task2 = await client.index(index2.uid).addDocuments([{ id: 1 }])
+      await client.waitForTask(task1.taskUid)
+      await client.waitForTask(task2.taskUid)
+
+      const tasks = await client.getTasks({
+        indexUid: [index.uid, index2.uid],
+      })
+      const onlyTaskWithSameUid = new Set(
+        tasks.results.map((task) => task.indexUid)
+      )
+
+      expect(onlyTaskWithSameUid.size).toEqual(2)
+    })
+
+    test(`${permission} key: Get all indexes tasks with index instance`, async () => {
+      const client = await getClient(permission)
+      const enqueuedTask = await client
+        .index(index.uid)
+        .addDocuments([{ id: 1 }])
+      await client.waitForTask(enqueuedTask.taskUid)
+
+      const tasks = await client.index(index.uid).getTasks()
+
+      expect(tasks.results[0]).toHaveProperty(
+        'status',
+        TaskStatus.TASK_SUCCEEDED
+      )
+      expect(tasks.results[0].indexUid).toEqual(index.uid)
```

Comment:
The same comment as before, I think here you just want to ensure that they belong to the same index name :)